test/org/javarosa/core/util/GeoShapeAreaTest.java

```diff
@@ -36,20 +41,34 @@
 public class GeoShapeAreaTest {
     private static final Logger logger = LoggerFactory.getLogger(GeoShapeAreaTest.class);
 
-  @Test public void testGeoShapeSupportForEnclosedArea() throws Exception {
-    // Read the form definition
-    final FormDef formDef = parse(r("area.xml")).formDef;
+    @Test public void testGeoShapeSupportForEnclosedArea() throws Exception {
+        // Read the form definition
+        final FormDef formDef = parse(r("area.xml")).formDef;
 
-    // Trigger all calculations
-    formDef.initialize(true, new InstanceInitializationFactory());
+        // Trigger all calculations
+        formDef.initialize(true, new InstanceInitializationFactory());
 
-    // Check the results. The data and expected results come from GeoUtilsTest.
-    TreeElement root = formDef.getMainInstance().getRoot();
+        // Check the results. The data and expected results come from GeoUtilsTest.
+        TreeElement root = formDef.getMainInstance().getRoot();
 
-    IAnswerData area = root.getChildAt(1).getValue();
-    assertEquals(151_452, (Double) area.getValue(), 0.5);
+        IAnswerData area = root.getChildAt(1).getValue();
+        assertEquals(151_452, (Double) area.getValue(), 0.5);
 
-    IAnswerData distance = root.getChildAt(2).getValue();
-    assertEquals(1_801, (Double) distance.getValue(), 0.5);
-  }
+        IAnswerData distance = root.getChildAt(2).getValue();
+        assertEquals(1_801, (Double) distance.getValue(), 0.5);
+    }
+
+    @Test
+    public void testAreaWithLessThanThreePoints() throws Exception {
+        FormDef formDef = parse(r("area_with_less_than_three_points.xml")).formDef;
+        try {
+            formDef.initialize(true, new InstanceInitializationFactory());
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof XPathUnhandledException) {
+                assertThat(e.getCause().getMessage(), containsString(REQUIRES_THREE_POINTS_MESSAGE));
+                return;
+            }
+        }
+        fail("The expected exception with error about three points being required was not thrown");
+    }
```

Comment:
![screen shot 2018-06-27 at 6 42 42 pm](https://user-images.githubusercontent.com/55099/42008381-114406e2-7a3a-11e8-8f69-90bc767146fb.png)

Two things:
1) I know your IDE alerts you to things that should be fixed. Does it show you that you created a duplicated block of non-trivial code?
2) I wrote and tested this nice code for you, yet git says you wrote it. Did you cherry pick it?

Reply:
1. I can't see such a prompt.
2. Fixed.

Reply:
For 1, I’ll be glad to help you remove the duplication if you like. Thanks for 2.

Reply:
Here’s what I’m thinking, to remove the duplication:

https://github.com/dcbriccetti/javarosa/commit/0d7f526b57ea5276ffd3ef7c7c06f05ce9353ad6

Reply:
Ok, I think it's ready.